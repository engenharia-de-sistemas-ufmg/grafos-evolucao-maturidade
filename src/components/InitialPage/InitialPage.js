import React from 'react';
import PropTypes from 'prop-types';

import basicGraph from '../../examples/basicGraph';
import systemEngineerGraph from '../../examples/systemEngineerGraph';

import * as labels from '../../commons/labels';

const InitialPage = ({ language, loadPage, updateGraph }) => {
  const handleBasicGraph = () => {
    updateGraph(basicGraph);
    loadPage();
  };

  const handleSystemEngineer = () => {
    updateGraph(systemEngineerGraph);
    loadPage();
  };

  return (
    <div>
      <div>{labels.initialPage[language]}</div>
      <button type="button" className="btn-basic-graph" onClick={handleBasicGraph}>
        {labels.basicModel[language]}
      </button>
      <button type="button" className="btn-system-engineer" onClick={handleSystemEngineer}>
        {labels.systemEngineer[language]}
      </button>
    </div>
  );
};

InitialPage.propTypes = {
  language: PropTypes.string.isRequired,
  loadPage: PropTypes.func.isRequired,
  updateGraph: PropTypes.func.isRequired,
};

export default InitialPage;
