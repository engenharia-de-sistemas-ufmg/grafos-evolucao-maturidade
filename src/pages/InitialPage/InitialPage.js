import React from 'react';
import PropTypes from 'prop-types';
import basicGraph from '../../examples/basicGraph';
import systemEngineerGraph from '../../examples/systemEngineerGraph';
import { defaultNodeGroups } from '../../commons/consts';
import * as labels from '../../commons/labels';

const InitialPage = ({
  lang, updateNodeGroups, updateGraph, viewGraph,
}) => {
  const click = (graph) => {
    updateNodeGroups(defaultNodeGroups);
    updateGraph(graph);
    viewGraph();
  };

  const title = (
    <header className="bd-header">
      <h1 className="title">
        {labels.maturityEvolutionGraphsTitle[lang]}
      </h1>
      <h2 className="subtitle">
        {labels.maturityEvolutionGraphsSubtitle[lang]}
      </h2>
      <p>&nbsp;</p>
    </header>
  );

  const button = (label, subtitle, buttonClass, graph) => (
    <div className="box">
      <article className="media">
        <div className="media-left">
          <button
            type="button"
            className={`${buttonClass} button is-primary`}
            onClick={() => {
              click(graph);
            }}
          >
            {label}
          </button>
        </div>
        <div className="media-content">
          {subtitle}
        </div>
      </article>
    </div>
  );

  const buttonBasicGraph = button(labels.basicModel[lang],
    labels.basicModelSubtitle[lang],
    'btn-basic-graph',
    basicGraph);

  const buttonSystemEngineer = button(labels.systemEngineer[lang],
    labels.systemEngineerSubtitle[lang],
    'btn-system-engineer',
    systemEngineerGraph);

  return (
    <section className="section">
      <div className="container">
        <div className="notification">
          {title}
          {buttonBasicGraph}
          {buttonSystemEngineer}
        </div>
      </div>
    </section>
  );
};

InitialPage.propTypes = {
  lang: PropTypes.string.isRequired,
  updateNodeGroups: PropTypes.func.isRequired,
  updateGraph: PropTypes.func.isRequired,
  viewGraph: PropTypes.func.isRequired,
};

export default InitialPage;
