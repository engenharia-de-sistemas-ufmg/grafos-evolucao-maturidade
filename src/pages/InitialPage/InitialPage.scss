.is-right {
  float: right;
}

.graph-example-container {
  display: flex;
  margin-bottom: 12px;
}

.graph-example-container > button {
  display: block;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 0;
  padding-right: 8px;
  width: 25%;
  height: auto;
  text-align: center;
  white-space: normal;
}

.graph-example-container > p {
  display: block;
  flex-basis: auto;
  flex-grow: 1;
  flex-shrink: 1;
  padding-left: 8px;
  width: 75%;
}
