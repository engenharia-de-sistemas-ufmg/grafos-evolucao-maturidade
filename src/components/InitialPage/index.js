import { connect } from 'react-redux';
import InitialPage from './InitialPage';
import { updateGraph, loadPage } from '../../actions';

const mapDispatchToProps = (dispatch) => ({
  loadPage: () => { dispatch(loadPage()); },
  updateGraph: (graph) => { dispatch(updateGraph(graph)); },
});

export default connect(
  null,
  mapDispatchToProps,
)(InitialPage);
