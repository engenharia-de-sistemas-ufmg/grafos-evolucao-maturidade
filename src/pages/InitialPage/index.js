import { connect } from 'react-redux';
import InitialPage from './InitialPage';
import { updateGraph } from '../../redux/graph.actions';
import { viewGraph, setFeatures } from '../../redux/page.actions';
import { updateNodeGroups, updateOptions } from '../../redux/general.actions';
import basicGraph from '../../examples/basicGraph';
import systemEngineerGraph from '../../examples/systemEngineerGraph';
import systemEngineerGraph20182 from '../../examples/systemEngineerGraph.20182';
import systemEngineerGraph20201 from '../../examples/systemEngineerGraph.20201';
import devOpsGraphGeneral from '../../examples/devOpsGraph.general';
import devOpsGraphDetailed from '../../examples/devOpsGraph.detailed';

const mapDispatchToProps = (dispatch) => ({
  updateNodeGroups: (nodeGroups) => { dispatch(updateNodeGroups(nodeGroups)); },
  updateGraph: (graph) => { dispatch(updateGraph(graph)); },
  viewGraph: () => { dispatch(viewGraph()); },

  openBasicGraph: () => {
    dispatch(setFeatures({ nexts: false }));
    dispatch(updateOptions({
      hierarchical: true,
      hierarchicalDirection: 'R',
      animation: true,
    }));
    dispatch(updateNodeGroups(basicGraph.groups));
    dispatch(updateGraph({ nodes: basicGraph.nodes, edges: basicGraph.edges }));
    dispatch(viewGraph());
  },

  openSystemEngineerGraph1: () => {
    dispatch(setFeatures({ nexts: false }));
    dispatch(updateOptions({
      hierarchical: true,
      hierarchicalDirection: 'R',
      animation: false,
    }));
    dispatch(updateNodeGroups(systemEngineerGraph.groups));
    dispatch(updateGraph({ nodes: systemEngineerGraph.nodes, edges: systemEngineerGraph.edges }));
    dispatch(viewGraph());
  },

  openSystemEngineerGraph2: () => {
    dispatch(setFeatures({ nexts: false }));
    dispatch(updateOptions({
      hierarchical: true,
      hierarchicalDirection: 'R',
      animation: false,
    }));
    dispatch(updateNodeGroups(systemEngineerGraph20182.groups));
    dispatch(updateGraph({
      nodes: systemEngineerGraph20182.nodes.map((n) => (
        { ...n, status: undefined })),
      edges: systemEngineerGraph20182.edges,
    }));
    dispatch(viewGraph());
  },

  openSystemEngineerGraph3: () => {
    dispatch(setFeatures({ nexts: true }));
    dispatch(updateOptions({
      hierarchical: true,
      hierarchicalDirection: 'R',
      animation: false,
    }));
    dispatch(updateNodeGroups(systemEngineerGraph20182.groups));
    dispatch(updateGraph({
      nodes: systemEngineerGraph20182.nodes,
      edges: systemEngineerGraph20182.edges,
    }));
    dispatch(viewGraph());
  },

  openSystemEngineerGraph4: () => {
    dispatch(setFeatures({ nexts: true }));
    dispatch(updateOptions({
      hierarchical: true,
      hierarchicalDirection: 'R',
      animation: false,
    }));
    dispatch(updateNodeGroups(systemEngineerGraph20182.groups));
    dispatch(updateGraph({
      nodes: systemEngineerGraph20182.nodes.map((n) => (
        { ...n, status: 0 })),
      edges: systemEngineerGraph20182.edges,
    }));
    dispatch(viewGraph());
  },

  openSystemEngineerGraph5: () => {
    dispatch(setFeatures({ nexts: true }));
    dispatch(updateOptions({
      hierarchical: true,
      hierarchicalDirection: 'R',
      animation: false,
    }));
    dispatch(updateNodeGroups(systemEngineerGraph20201.groups));
    dispatch(updateGraph({
      nodes: systemEngineerGraph20201.nodes.map((n) => (
        { ...n, status: 2 })),
      edges: systemEngineerGraph20201.edges,
    }));
    dispatch(viewGraph());
  },

  openDevOpsGraph1: () => {
    dispatch(setFeatures({ nexts: false }));
    dispatch(updateOptions({
      hierarchical: true,
      hierarchicalDirection: 'D',
      animation: true,
    }));
    dispatch(updateNodeGroups(devOpsGraphGeneral.groups));
    dispatch(updateGraph({
      nodes: devOpsGraphGeneral.nodes,
      edges: devOpsGraphGeneral.edges,
    }));
    dispatch(viewGraph());
  },

  openDevOpsGraph2: () => {
    dispatch(setFeatures({ nexts: false }));
    dispatch(updateOptions({
      hierarchical: false,
      animation: true,
    }));
    dispatch(updateNodeGroups(devOpsGraphDetailed.groups));
    dispatch(updateGraph({
      nodes: devOpsGraphDetailed.nodes.map((n) => (
        { ...n, status: undefined })),
      edges: devOpsGraphDetailed.edges,
    }));
    dispatch(viewGraph());
  },

  openDevOpsGraph3: () => {
    dispatch(setFeatures({ nexts: false }));
    dispatch(updateOptions({
      hierarchical: false,
      animation: true,
    }));
    dispatch(updateNodeGroups(devOpsGraphDetailed.groups));
    dispatch(updateGraph({
      nodes: devOpsGraphDetailed.nodes,
      edges: devOpsGraphDetailed.edges,
    }));
    dispatch(viewGraph());
  },
});

export default connect(
  null,
  mapDispatchToProps,
)(InitialPage);
