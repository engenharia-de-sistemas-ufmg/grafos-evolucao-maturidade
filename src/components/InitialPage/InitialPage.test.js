import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import InitialPage from './InitialPage';

import basicGraph from '../../examples/basicGraph';
import systemEngineerGraph from '../../examples/systemEngineerGraph';

describe('InitialPage', () => {
  Enzyme.configure({ adapter: new Adapter() });

  test('should match snapshot', () => {
    const lang = 'pt';
    const loadPage = () => { };
    const updateGraph = () => { };
    const component = shallow(
      <InitialPage
        lang={lang}
        loadPage={loadPage}
        updateGraph={updateGraph}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  test('should trigger basicGraph', () => {
    const lang = 'pt';
    const loadPage = sinon.spy();
    const updateGraph = sinon.spy();
    const component = shallow(
      <InitialPage
        lang={lang}
        loadPage={loadPage}
        updateGraph={updateGraph}
      />,
    );
    component.find('.btn-basic-graph').simulate('click');
    expect(loadPage.callCount).toBe(1);
    expect(updateGraph.callCount).toBe(1);
    expect(updateGraph.args[0][0]).toBe(basicGraph);
  });

  test('should trigger systemEngineer', () => {
    const lang = 'pt';
    const loadPage = sinon.spy();
    const updateGraph = sinon.spy();
    const component = shallow(
      <InitialPage
        lang={lang}
        loadPage={loadPage}
        updateGraph={updateGraph}
      />,
    );
    component.find('.btn-system-engineer').simulate('click');
    expect(loadPage.callCount).toBe(1);
    expect(updateGraph.callCount).toBe(1);
    expect(updateGraph.args[0][0]).toBe(systemEngineerGraph);
  });
});
