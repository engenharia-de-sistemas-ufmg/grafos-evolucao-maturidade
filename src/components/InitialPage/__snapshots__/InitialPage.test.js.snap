// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InitialPage should match snapshot 1`] = `
<div>
  <div>
    Página inicial
  </div>
  <button
    className="btn-basic-graph"
    onClick={[Function]}
    type="button"
  >
    Modelo básico
  </button>
  <button
    className="btn-system-engineer"
    onClick={[Function]}
    type="button"
  >
    Engenharia de Sistemas
  </button>
</div>
`;
